scripts/esbuild/build.mjs

```diff
@@ -0,0 +1,146 @@
+import { existsSync, mkdirSync } from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { stripNodePrefixPlugin } from "./build/plugins/stripNodePrefixPlugin.mjs";
+import { ignoreModulesPlugin } from "./build/plugins/ignoreModulesPlugin.mjs";
+import { postBuildPlugin } from "./build/plugins/postBuildPlugin.mjs";
+import { DIST_FOLDER, NODE_IGNORE_MODULES, NODE_EXTERNAL_MODULES, WEB_IGNORE_MODULES, WEB_EXTERNAL_MODULES } from "./build/constants.mjs";
+import { reportSize } from "./build/reportSize.mjs";
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+const rootDir = path.join(__dirname, "../..");
+
+/**
+ * Helper function to create build configurations.
+ * Equivalent to webpack's buildConfig function.
+ */
+async function buildTarget({
+  name = "",
+  suffix = ".js",
+  format = "esm", // 'esm' | 'cjs'
+  ignoreModules = [],
+  externalModules = [],
+  usePostBuild = false,
+}) {
+  const platform = format === "cjs" ? "node" : "neutral";
+  const outdir = path.join(rootDir, DIST_FOLDER);
+
+  if (!existsSync(outdir)) {
+    mkdirSync(outdir, { recursive: true });
+  }
+
+  const regularFile = `transformers${name}${suffix}`;
+  const minFile = `transformers${name}.min${suffix}`;
+
+  const plugins = [];
+  // Add ignoreModulesPlugin FIRST so it can catch modules before stripNodePrefixPlugin marks them as external
+  if (ignoreModules.length > 0) {
+    plugins.push(ignoreModulesPlugin(ignoreModules));
+  }
+  plugins.push(stripNodePrefixPlugin());
+  if (usePostBuild) {
+    plugins.push(postBuildPlugin(outdir, rootDir));
+  }
+
+  console.log(`\nBuilding ${regularFile}...`);
+  await esbuild({
+    bundle: true,
+    treeShaking: true,
+    logLevel: "warning",
+    entryPoints: [path.join(rootDir, "src/transformers.js")],
+    platform,
+    format,
+    outfile: path.join(outdir, regularFile),
+    sourcemap: true,
+    external: externalModules,
+    plugins,
+    logOverride: {
+      // Suppress import.meta warning for CJS builds - it's handled gracefully in the code
+      "empty-import-meta": "silent",
+    },
+  });
+  reportSize(path.join(outdir, regularFile));
+
+  console.log(`\nBuilding ${minFile}...`);
+  await esbuild({
+    bundle: true,
+    treeShaking: true,
+    logLevel: "warning",
+    entryPoints: [path.join(rootDir, "src/transformers.js")],
+    platform,
+    format,
+    outfile: path.join(outdir, minFile),
+    sourcemap: true,
+    minify: true,
+    external: externalModules,
+    plugins,
+    legalComments: "none",
+    logOverride: {
+      // Suppress import.meta warning for CJS builds - it's handled gracefully in the code
+      "empty-import-meta": "silent",
+    },
+  });
+  reportSize(path.join(outdir, minFile));
+}
+
+console.log("\nBuilding transformers.js with esbuild...\n");
+
+const startTime = performance.now();
+
+try {
+  console.log("=== CLEAN ===");
+  execSync(`rimraf ${DIST_FOLDER}`, { stdio: "inherit" });
```

Comment:
if we are going to call rimraf using execSync, we might as well just use native fs functions to clear the dist directory. -1 dependency :)